patches/server/0934-Configurable-region-compression-format.patch

```diff
@@ -0,0 +1,45 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Peter Crawley <peterzcrawley@gmail.com>
+Date: Thu, 27 Oct 2022 22:19:31 +0100
+Subject: [PATCH] Configurable region compression format
+
+
+diff --git a/src/main/java/net/minecraft/world/level/chunk/storage/RegionFile.java b/src/main/java/net/minecraft/world/level/chunk/storage/RegionFile.java
+index fc3442b4c7e1f22080fe6bf36d4fade162d6709e..3fb3f8b1a3da79f97c64e1ff631c59bff514707e 100644
+--- a/src/main/java/net/minecraft/world/level/chunk/storage/RegionFile.java
++++ b/src/main/java/net/minecraft/world/level/chunk/storage/RegionFile.java
+@@ -425,11 +425,11 @@ public class RegionFile implements AutoCloseable {
+     // Paper end
+ 
+     public RegionFile(Path file, Path directory, boolean dsync) throws IOException {
+-        this(file, directory, RegionFileVersion.VERSION_DEFLATE, dsync);
++        this(file, directory, RegionFileVersion.compressionFormatSupplier.get(), dsync); // Paper - Configurable region compression format
+     }
+     // Paper start - add can recalc flag
+     public RegionFile(Path file, Path directory, boolean dsync, boolean canRecalcHeader) throws IOException {
+-        this(file, directory, RegionFileVersion.VERSION_DEFLATE, dsync, canRecalcHeader);
++        this(file, directory, RegionFileVersion.compressionFormatSupplier.get(), dsync, canRecalcHeader); // Paper - Configurable region compression format
+     }
+     // Paper end - add can recalc flag
+ 
+diff --git a/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileVersion.java b/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileVersion.java
+index 97d4ae5619dcc0922e0381b1bb45a135f514e3af..78eec7232e223732d415e70b47f1f4302929658c 100644
+--- a/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileVersion.java
++++ b/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileVersion.java
+@@ -30,6 +30,16 @@ public class RegionFileVersion {
+     }, (outputStream) -> {
+         return outputStream;
+     }));
++
++    // Paper Start - Configurable region compression format
++    public static final java.util.function.Supplier<RegionFileVersion> compressionFormatSupplier = () ->
```

Comment:
Use ALL_CAPS for the field name here since it’s a static final field.

Reply:
Done.